packages/es-components/src/components/controls/buttons/SpinnerButton.js

```diff
@@ -0,0 +1,97 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import Button from './Button';
+import Spinner from '../../base/spinner/Spinner';
+
+const DISPLAY_TYPES = {
+  none: 'none',
+  inline: 'inline'
+};
+
+function SpinnerButton({
+  children,
+  onClick,
+  styleType,
+  size,
+  block,
+  mobileBlock,
+  flatLeftEdge,
+  flatRightEdge,
+  title,
+  description,
+  ...other
+}) {
+  const [displaySpinner, setDisplay] = useState(DISPLAY_TYPES.none);
+
+  const defaultDimensions = {
+    maxWidth: 100,
+    height: 40
+  };
+
+  const loadSpinnerTriggerOnClick = () => {
+    setDisplay(DISPLAY_TYPES.inline);
+    const onClickPromise = new Promise(onClick);
+
+    onClickPromise.then(
+      setTimeout(() => {
+        setDisplay(DISPLAY_TYPES.none);
+      }, 500)
+    );
+  };
+
+  return (
+    <Button
+      size={size}
+      block={block}
+      mobileBlock={mobileBlock}
+      flatLeftEdge={flatLeftEdge}
+      flatRightEdge={flatRightEdge}
+      onClick={loadSpinnerTriggerOnClick}
+      styleType={styleType}
+      style={defaultDimensions}
+      {...other}
+    >
+      {displaySpinner === 'inline' ? (
+        <Spinner
+          height="70%"
+          width="70%"
```

Comment:
How does this look in mobile?

Reply:
Hey, I just added screenshots above. Thanks for this question.